experimental/sdk/webpubsub-graphql-subscribe/src/WpsWebSocketServer.ts

```diff
@@ -0,0 +1,135 @@
+
+import { WebPubSubServiceClient } from "@azure/web-pubsub";
+import { WebPubSubEventHandler } from "@azure/web-pubsub-express";
+import { ApolloServer} from "apollo-server-express";
+import { execute, GraphQLSchema, subscribe } from "graphql";
+import { SubscriptionServer } from 'subscriptions-transport-ws';
+import express from 'express';
+import WebSocket from "ws";
+import { WpsPubSub } from "./azure-wps-pubsub";
+import {LOG, log, config} from "./utils"
+
+/**
+  * a virtual WebSocket.Server without real server
+ */
+class VirtualWebSocketServer extends WebSocket.Server{
+	protocol: string = "graphql-ws";
+	readyState: number = WebSocket.OPEN;
+
+	constructor() {
+		 super({port: undefined, noServer: true}); 
+	}
+}
+
+/** 
+  * A VirtualWebSocketServer which is binded to an Azure Web PubSub client connection according to `servecClient` and `connectionId`
+  * It overrides method `send` and send all message to its binded connection in JSON format.
+ */
+class SubWpsWebSocketServer extends VirtualWebSocketServer{
+	serviceClient: WebPubSubServiceClient;
+	connectionId: string;
+
+	constructor(serviceClient: WebPubSubServiceClient, connectionId: string) {
+		super();
+		this.serviceClient = serviceClient;
+		this.connectionId = connectionId;
+	}
+
+	@LOG("[send] [SubWsServer]")
+	send(data: any) { this.serviceClient.sendToConnection(this.connectionId, JSON.parse(data)); }
```

Comment:
you can use `this.serviceClient.sendToConnection(this.connectionId, data, { contentType: "text/plain" })`

Reply:
fixed